extensions/blocks/eventbrite/edit.js

```diff
@@ -0,0 +1,379 @@
+		resolvedStatusCode: null,
+	};
+
+	componentDidMount() {
+		const { resolvingUrl } = this.state;
+
+		// Check if we need to resolve an Eventbrite URL immediately.
+		if ( resolvingUrl ) {
+			this.resolveUrl();
+		}
+	}
+
+	componentDidUpdate( prevProps, prevState ) {
+		// Check if an Eventbrite URL has been entered, so we need to resolve it.
+		if ( ! prevState.resolvingUrl && this.state.resolvingUrl ) {
+			this.resolveUrl();
+		}
+	}
+
+	// TODO: figure out how to cancel request since apiFetch was updated to use Promises rather than XHR requests.
+	// componentWillUnmount() {}
+
+	resolveUrl = () => {
+		const { url } = this.props.attributes;
+
+		this.setState( { resolvedStatusCode: null } );
+
+		this.fetchRequest = apiFetch( {
+			path: `/wpcom/v2/resolve-redirect/${ url }`,
+		} );
+
+		this.fetchRequest.then(
+			response => {
+				// resolve
+				this.fetchRequest = null;
+				const resolvedUrl = response.url || url;
+				const resolvedStatusCode = response.status ? parseInt( response.status, 10 ) : null;
+
+				this.props.setAttributes( { url: resolvedUrl } );
+				this.setState( {
+					resolvingUrl: false,
+					resolvedStatusCode,
+					editedUrl: resolvedUrl,
+				} );
+			},
+			xhr => {
+				// reject
+				if ( xhr.statusText === 'abort' ) {
+					return;
+				}
+				this.fetchRequest = null;
+				this.setState( {
+					resolvingUrl: false,
+					editingUrl: true,
+				} );
+			}
+		);
+	};
+
+	setUrl = event => {
+		if ( event ) {
+			event.preventDefault();
+		}
+
+		const { editedUrl: url } = this.state;
+
+		if ( ! url ) {
+			return;
+		}
+
+		this.props.setAttributes( { url } );
+
+		// Setting the `resolvingUrl` state here, then waiting for `componentDidUpdate()` to
+		// be called before actually resolving it ensures that the `editedUrl` state has also been
+		// updated before resolveUrl() is called.
+		this.setState( {
+			editingUrl: false,
+			resolvingUrl: true,
+		} );
+	};
+
+	cannotEmbed = () => {
+		const { url } = this.props.attributes;
+		const { resolvedStatusCode } = this.state;
+
+		return (
+			( url && ! URL_REGEX.test( url ) ) || ( resolvedStatusCode && resolvedStatusCode >= 400 )
+		);
+	};
+
+	setEmbedType = embedType => {
+		const { setAttributes } = this.props;
+
+		setAttributes( { useModal: 'modal' === embedType } );
+	};
+
+	renderLoading() {
+		return (
+			<div className="wp-block-embed is-loading">
+				<Spinner />
+				<p>{ __( 'Embedding…' ) }</p>
+			</div>
+		);
+	}
+
+	renderInspectorControls() {
+		const { useModal } = this.props.attributes;
+		const embedTypes = [
+			{
+				value: 'inline',
+				isActive: ! useModal,
+				label: __( 'In-page Embed', 'jetpack' ),
+				image: EventbriteInPageExample,
+				alt: __( 'In page Eventbrite checkout example', 'jetpack' ),
+			},
+			{
+				value: 'modal',
+				isActive: useModal,
+				label: __( ' Button & Modal', 'jetpack' ),
+				image: EventbriteModalExample,
+				alt: __( 'Modal Eventbrite checkout example', 'jetpack' ),
+			},
+		];
```

Comment:
The example images are too small when displayed on mobile (at the largest mobile width):

<img width="772" alt="Screenshot 2020-01-16 at 13 55 57" src="https://user-images.githubusercontent.com/2070010/72530867-35e25100-3868-11ea-9da7-7110e3926dbe.png">

We could either provide larger images (if I'm not mistaken, the container would be ~355px wide), or have them 100% wide in CSS:
```css
.block-editor__container img {
  width: 100%;
}
```

One other thing I've noticed is that the primary button in my in-page embed is green, while in the example is orange.
The button embed too has a different color, but in that case it feels to me less awkward as I can customize it as I wish.

Reply:
I've updated the image size for the in-page embed and changed the button version to be a kind of live preview, what do you think of that?

Reply:
I marked this as resolved and then unresolved because otherwise it might get lost in the amount of comments.
I like this solution more, and it's definitely more Gutenberg-y.

There's an issue with it, though: the preview does not automatically resize with the sidebar width.
This can lead to awkward positioning/sizes of the preview, but it's also a Core issue (e.g. the Separator styles suffer the same problem).

You can try it yourself by resizing to a mobile width, opening the sidebar on the Eventbrite block, and then resize the window again (the sidebar should remain open when going to a desktop width).

Just to clarify: I'm just pointing out that this issue exists, but I'm not considering it a blocker.